### PR TITLE
Rentals update

### LINKED ISSUE #51: rentals controller needs to be rewritten for updated api

### PR DESCRIPTION
**Pull request by tomecho**
fixes #51

### CODE REVIEW

File: app/models/rental.rb [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+
+  before_create :create_reservation
+  before_destroy :delete_reservation
 
   belongs_to :user
   belongs_to :department
   belongs_to :item_type
 
-  validates :reservation_id, presence: true, unless: :skip_reservation_validation
+  # validate :reservation_id, presence: true, unless: :skip_reservation_validation
```
Comment (tomecho) [inline app/models/rental.rb]: is this still necessary?

File: app/controllers/rentals_controller.rb [hunk 2]
```diff
@@ -27,25 +27,20 @@ def create
...
   # DELETE /rentals/1
   def destroy
-    if @rental.delete_reservation
-      @rental.destroy
-      flash[:success] = 'Rental Was Successfully Deleted'
-    else
-      @rental.errors.full_messages.each { |e| flash_message :warning, e, :now }
-    end
+    @rental.destroy
+    flash[:success] = 'Rental Was Successfully Deleted'
```
Comment (dfaulken) [inline app/controllers/rentals_controller.rb]: Stylistic point: it's a little awkward to titleize messages

File: app/models/inventory.rb [hunk 3]
```diff
@@ -15,20 +17,21 @@ def self.create_item_type(name, allowed_keys = [])
                              body: { 'name' => name, 'allowed_keys' => allowed_keys }.to_json,
                              headers: @post_headers)
     handle_item_type_errors(response)
-    JSON.parse(response.body)
+    JSON.parse(response.body).with_indifferent_access
```
Comment (dfaulken) [inline app/models/inventory.rb]: Sweet, didn't know you could do that

File: app/models/rental.rb [hunk 4]
```diff
@@ -48,23 +52,30 @@ class Rental < ActiveRecord::Base
...
-
-    errors.add(:base, 'Error occured in aggressive epsilon: unable to delete reservation')
-    false
+    return true if reservation_id.nil? # nothing to delete here
+    return true if end_date < Time.current # deleting it is pointless, it wont inhibit new rentals and it will destroy a record.
+    begin
+      Inventory.delete_reservation(reservation_id)
+      self.reservation_id = nil
+    rescue => error
+      errors.add(:base, error.inspect) && (return false)
```
Comment (dfaulken) [inline app/models/rental.rb]: I think `and return false` will work just as well.

**Comment by tomecho** [comment]
once the pull for aggressive epsilon is merged ill get travis to pass

**Pull request merged**